src/useValidation/useValidation.ts

```diff
@@ -83,21 +83,18 @@ export function useValidation<T>(
         previousValue: previousValueRef.value,
       })
 
-      if (result === undefined) {
-        return valid.value
-      }
-
       error.value = result
+      pending.value = false
+      validated.value = true
+      previousValueRef.value = valueRef.value
```

Comment:
Makes sure these only get set if validation is not aborted. Otherwise it can appear that validation is complete sooner than it actually is.